proto/proto-backplane-grpc/src/main/proto/deephaven/proto/storage.proto

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+syntax = "proto3";
+
+package io.deephaven.proto.backplane.grpc;
+
+option java_multiple_files = true;
+option optimize_for = SPEED;
+option go_package = "github.com/deephaven/deephaven-core/go/internal/proto/storage";
+
+/*
+ * Shared storage management service.
+*/
+service StorageService {
+  rpc ListItems(ListItemsRequest) returns (ListItemsResponse) {}
+  rpc FetchFile(FetchFileRequest) returns (FetchFileResponse) {}
```

Comment:
We might want this to be a streaming endpoint for very large files, see also the ByteRange message (commented out) by which a client could request regions of a file.

If we go down this road we would likely want a way to update ranges in a file too, appending new ranges, truncating the file, etc.

Reply:
Yeah, I'm thinking about something like a user uploading/downloading a 10gb parquet file to the server

Reply:
I do wonder - are we trying to bite off too general of use case by trying to imagine this as a service for large binary blobs? At least in the case of notebooks, the use case is more specifically targeted towards "support script/text files and hierarchical management of said files".

I think there is potential for confusion where this feature gets overly pigeon-holed as a disk-backed local-storage available to the server process. IE, your example - a user uploads a 10GB parquet file to the *service* and now is confused why it's not available on the local disk.

In which case, we may need to have some sort support for resolving _StorageService_ URIs - something that may be local, or may be remote. But then we may run into a lot of NFS-like issues, where we ask "why is opening a parquet file so slow?"

Note: I'm not against the binary blob use case, but I think we need to think off all that it might entail.

Reply:
Depending on the layers of abstraction, this sort of issue might already be present - for example, remote docker or kubernetes volumes masquerading as disk-backed storage.

There *may* be room for us to establish a convention where werecommend setting up some (or all) of the docker server volumes as `StorageService`. This might also mean we should have ways of starting *multiple* `StorageService` for the same deployment.

Reply:
Biting off too much: Agreed, I don't want to solve every possible problem, but I do want to at least have a plan on how we might solve other issues, to avoid changing the api later (for example, making FetchFile server-streaming or SaveFile client-streaming). 

Multi-service: this is an explicit design goal for this and the config service, though the discovery mechanism isn't defined yet. gRPC itself disallows (well, technically "strongly discourages" prefixing URLs with additional path segments, but we are effectively free to add other request metadata or fields to the message to clarify "which" service a client is talking to. The `path` field may itself an effective way to communicate which sets of storage are available (by convention) in a single server, but with any kind of load balancing or discovery (even as simple as just different hosts/ports, etc) we can easily build on this simple service to offer customized setups.

My initial suggestion is to keep this and the config service light and flexible, and encourage requesting configuration first to establish specific path prefixes that are interesting - such as "where are notebooks", "where are layouts".

Reply:
With a forward looking, I'd be in favor of a streaming response here - I don't think we need any additional metadata on the response. The contract being for now that you append all the bytes together to get the full file.

Reply:
A streaming "fetch" requires a matching multi-call "save", or you can only download and not upload. Make save client-upload, and wire up our fake streaming upload feature for gRPC-web?